community/kernel/src/main/java/org/neo4j/kernel/impl/recovery/RecoveryRequiredChecker.java

```diff
@@ -66,8 +65,7 @@ public boolean isRecoveryRequiredAt( File dataDir ) throws IOException
         PhysicalLogFiles logFiles = new PhysicalLogFiles( dataDir, fs );
 
         LogEntryReader<ReadableClosablePositionAwareChannel> reader =
-                new VersionAwareLogEntryReader<>( LogEntryVersion.CURRENT.byteCode(),
-                        new RecordStorageCommandReaderFactory() );
+                new VersionAwareLogEntryReader<>( new RecordStorageCommandReaderFactory() );
```

Comment:
`VersionAwareLogEntryReader` is always constructed in the same way. Maybe add a default 0-arg constructor with creates or uses singleton `RecordStorageCommandReaderFactory` inside?